modules/andes-core/broker/src/main/java/org/wso2/andes/mqtt/connectors/PersistenceStoreConnector.java

```diff
@@ -94,6 +94,15 @@ public void addMessage(MQTTMessageContext messageContext) throws MQTTException {
                 publisherTopicCorrelate.put(messageContext.getPublisherID(), publisher);
                 //Finally will register the publisher channel for flow controlling
                 AndesChannel publisherChannel = Andes.getInstance().createChannel(publisher);
+                //Set channel details
+                //Substring to remove leading slash character from remote address
+                try {
+                    publisherChannel.setIdentifier(
+                            messageContext.getChannel().remoteAddress().toString().substring(1));
+                } catch (NullPointerException e) {
+                    publisherChannel.setIdentifier("MQTT-Unknown");
```

Comment:
IMO this should perform a null check and should not allow creation of the publisher or perhaps should trigger a warning. @AnujaLK  WDYT ? any idea why we're doing this ?